WordPress/Classes/Utility/ContextManager.m

```diff
@@ -203,95 +204,89 @@ - (void)mergeChanges:(NSManagedObjectContext *)context fromContextDidSaveNotific
     }];
 }
 
-- (BOOL)didMigrationFail
-{
-    return _migrationFailed;
-}
-
 
 #pragma mark - Setup
 
-- (NSManagedObjectModel *)managedObjectModel
-{
-    if (_managedObjectModel) {
-        return _managedObjectModel;
-    }
-    NSString *modelPath = [self modelPath];
-    NSURL *modelURL = [NSURL fileURLWithPath:modelPath];
-    _managedObjectModel = [[NSManagedObjectModel alloc] initWithContentsOfURL:modelURL];
-    return _managedObjectModel;
-}
-
-- (NSPersistentStoreCoordinator *)persistentStoreCoordinator
+- (NSPersistentContainer *)persistentContainer
 {
-    if (_persistentStoreCoordinator) {
-        return _persistentStoreCoordinator;
+    if (_persistentContainer) {
+        return _persistentContainer;
     }
```

Comment:
I'm wondering if it wouldn't be best to rely on GCD to initialize all the things *once* (rather than having lazy non threadsafe properties).

Reply:
Since we're going to change this whole code to Swift very soon, I guess we can tackle that on the next task.

What do you think?